core/src/net/sf/openrocket/rocketcomponent/LaunchLug.java

```diff
@@ -279,4 +287,49 @@ public AngleMethod getAngleMethod() {
 	public void setAngleMethod(AngleMethod newMethod) {
 		// no-op
 	}
+
+	@Override
+	public Appearance getInsideAppearance() {
+		return this.insideAppearance;
+	}
+
+	@Override
+	public void setInsideAppearance(Appearance appearance) {
+		this.insideAppearance = appearance;
+		if (this.insideAppearance != null) {
+			Decal d = this.insideAppearance.getTexture();
+			if (d != null) {
+				d.getImage().addChangeListener(new StateChangeListener() {
+
+					@Override
+					public void stateChanged(EventObject e) {
+						fireComponentChangeEvent(ComponentChangeEvent.TEXTURE_CHANGE);
+					}
+
+				});
+			}
+		}
+		// CHECK - should this be a TEXTURE_CHANGE and not NONFUNCTIONAL_CHANGE?
```

Comment:
I think the current event type is fine.

Reply:
> In general, love the change. Its clearly been thought through and implemented nicely.
> 
> However, I'm not in love with the copy-pasta between the different components - but I understand why you went this route and don't think its the worst thing in the world. I'd prefer to see something like an InsideAppearanceHandler or something to that effect that wraps all of the copy/paste for the setInsideAppearance. The reason is that in 3-6 months, we'll have a bug where it will get fixed one place but not the others.

100% agree, I just didn't see any other possibility other than an interface at the time, but a handler is a good idea!